src/main/scala/nl.knaw.dans.easy.sword2/DepositHandler.scala

```diff
@@ -110,16 +110,31 @@ object DepositHandler {
     }
 
     val payload = Paths.get(settings.tempDir.toString, id, deposit.getFilename.split("/").last).toFile
-    for {
+    val depositDir = Paths.get(settings.tempDir.toString, id).toFile
+    val receipt: Try[DepositReceipt] = extractPayloadAndGetDepositReceipt(deposit, contentLength, payload, depositDir)
+    setFilePermissions(settings, id, depositDir) // ensure all files in deposit are accessible for the deposit group
```

Comment:
looks like you're throwing away a potential `Failure` here... use a for-comprehension in combination with the line above.

Reply:
Yes, i was looking a that as well. I want that setFilesPermissions is always executed, whether extractPayloadAndGetDepositReceipt is a success or not

Reply:
that's interesting... because `Try` is failfast, as you know.
However, why do you want `setFilesPermissions` to be executed always? Is there any file to set the permissions for when `extractPayloadAndGetDepositReceipt` fails?

Reply:
It is a bit messy, during manual testing i found out it can leave a zip file, a bag with an invalid md5 or both.

Reply:
It's not very clean, but you could do something like this:

```scala
val x: Try[Int] = f() // f :: () -> Try[Int]
val y: Try[Unit] = g() // g :: () -> Try[Unit]

for {
  a <- x
  _ <- y
} yield a
```

by running `f` and `g` first, you perform their respective side effects first and capture the result (`Success`/`Failure`) in a value. Afterwards, you merge the 2 results using a for-comprehension. Not a clean way, but at least it's something...

In this case, document carefully why you do it this way (perhaps also refer to this discussion), because otherwise someone (probably me 😉) is going to refactor this!

Reply:
What you could also do is:

```scala
// f :: () -> Try[Int]
// g :: () -> Try[Unit]

f()
  .recoverWith { case e => g().flatMap(_ => Failure(e)) }
  .flatMap(i => g().map(_ => i))
```

this is probably a better solution!

Reply:
Or:

```scala
// f :: () -> Try[Int]
// g :: () -> Try[Unit]

f() match {
  case Success(i) => g().map(_ => i)
  case Failure(e) => g().flatMap(_ => Failure(e))
}
```

Reply:
think the last one is the most elegant

Reply:
thought so too. Looks great the way you did it!